src/main/scala/org/broadinstitute/dsde/firecloud/utils/StandardUserInfoDirectives.scala

```diff
@@ -14,10 +14,11 @@ trait StandardUserInfoDirectives extends UserInfoDirectives {
       headerValueByName("OIDC_CLAIM_user_id") &
       headerValueByName("OIDC_CLAIM_expires_in") &
       headerValueByName("OIDC_CLAIM_email") &
-      optionalHeaderValueByName("OAUTH2_CLAIM_google_id")
+      optionalHeaderValueByName("OAUTH2_CLAIM_google_id") &
+      optionalHeaderValueByName("OAUTH2_CLAIM_idp_access_token")
     ) tmap {
-      case (token, userId, expiresIn, email, googleIdOpt) => {
-        UserInfo(email, OAuth2BearerToken(token), expiresIn.toLong, googleIdOpt.getOrElse(userId))
+      case (token, userId, expiresIn, email, googleIdOpt, googleTokenOpt) => {
+        UserInfo(email, OAuth2BearerToken(token), expiresIn.toLong, googleIdOpt.getOrElse(userId), googleTokenOpt.map(OAuth2BearerToken))
```

Comment:
Just thinking something through. We're setting the `UserInfo.userId` here as `googleIdOpt.getOrElse(userId)`. Does this mean we can just test `UserInfo.isB2C` to test whether it's an azure user? So we don't really need to read the `OAUTH2_CLAIM_idp_access_token` header.

Note this logic to set the userId is kind of unique to Orch (Rawls [doesn't do it](https://github.com/broadinstitute/rawls/blob/develop/core/src/main/scala/org/broadinstitute/dsde/rawls/openam/StandardUserInfoDirectives.scala#L37)) -- the reason is essentially to make Thurloe backwards compatible with existing Google users through B2C.

Reply:
Oh interesting, that's a good point would simplify things a little bit. On the other hand, reading the `OAUTH2_CLAIM_idp_access_token` header is more in line with behavior of the other services and doesn't rely on a backwards compatibility quirk that might just be a temporary(?) solution.

I'm somewhat ambivalent here. Do you have any strong preference that I change it?

Reply:
Circling back to his, I don't have a strong opinion. I think the current version works, was just thinking if could be simplified. I can buy the argument that this way is more understandable.